database/query/profiles.sql

```diff
@@ -6,20 +6,47 @@ INSERT INTO profiles (
     alert,
     name) VALUES ($1, $2, $3, $4, $5) RETURNING *;
 
+-- name: UpdateProfile :one
+UPDATE profiles SET
+    remediate = $2,
+    alert = $3
+WHERE id = $1 RETURNING *;
+
 -- name: CreateProfileForEntity :one
 INSERT INTO entity_profiles (
     entity,
     profile_id,
     contextual_rules) VALUES ($1, $2, sqlc.arg(contextual_rules)::jsonb) RETURNING *;
 
+-- name: UpsertProfileForEntity :one
+INSERT INTO entity_profiles (
+    entity,
+    profile_id,
+    contextual_rules) VALUES ($1, $2, sqlc.arg(contextual_rules)::jsonb)
+ON CONFLICT (entity, profile_id) DO UPDATE SET
+    contextual_rules = sqlc.arg(contextual_rules)::jsonb
+RETURNING *;
+
+-- name: DeleteProfileForEntity :exec
+DELETE FROM entity_profiles WHERE profile_id = $1 AND entity = $2;
+
+-- name: GetProfileForEntity :one
+SELECT * FROM entity_profiles WHERE profile_id = $1 AND entity = $2;
+
 -- name: GetProfileByProjectAndID :many
 SELECT * FROM profiles JOIN entity_profiles ON profiles.id = entity_profiles.profile_id
 WHERE profiles.project_id = $1 AND profiles.id = $2;
 
 -- name: GetProfileByID :one
 SELECT * FROM profiles WHERE id = $1;
 
--- name: GetProfileByProjectAndName :many
+-- name: GetProfileByIDAndLock :one
+SELECT * FROM profiles WHERE id = $1 FOR UPDATE;
```

Comment:
Why did you include the `AndLock` in the name?

Reply:
because this `SELECT` actually has a `FOR UPDATE` suffix which locks the row.

Reply:
ah, thanks!